contracts/token/ERC20/ERC20Snapshot.sol

```diff
@@ -87,26 +89,27 @@ contract ERC20Snapshot is ERC20 {
         super._burn(account, value);
     }
 
-    // When a valid snapshot is queried, there are three possibilities:
-    //  a) The queried value was not modified after the snapshot was taken. Therefore, a snapshot entry was never
-    //  created for this id, and all stored snapshot ids are smaller than the requested one. The value that corresponds
-    //  to this id is the current one.
-    //  b) The queried value was modified after the snapshot was taken. Therefore, there will be an entry with the
-    //  requested id, and its value is the one to return.
-    //  c) More snapshots were created after the requested one, and the queried value was later modified. There will be
-    //  no entry for the requested id: the value that corresponds to it is that of the smallest snapshot id that is
-    //  larger than the requested one.
-    //
-    // In summary, we need to find an element in an array, returning the index of the smallest value that is larger if
-    // it is not found, unless said value doesn't exist (e.g. when all values are smaller). Arrays.findUpperBound does
-    // exactly this.
     function _valueAt(uint256 snapshotId, Snapshots storage snapshots)
         private view returns (bool, uint256)
     {
         require(snapshotId > 0, "ERC20Snapshot: id is 0");
         // solhint-disable-next-line max-line-length
         require(snapshotId <= _currentSnapshotId.current(), "ERC20Snapshot: nonexistent id");
 
+        // When a valid snapshot is queried, there are three possibilities:
+        //  a) The queried value was not modified after the snapshot was taken. Therefore, a snapshot entry was never
+        //  created for this id, and all stored snapshot ids are smaller than the requested one. The value that corresponds
+        //  to this id is the current one.
+        //  b) The queried value was modified after the snapshot was taken. Therefore, there will be an entry with the
+        //  requested id, and its value is the one to return.
+        //  c) More snapshots were created after the requested one, and the queried value was later modified. There will be
+        //  no entry for the requested id: the value that corresponds to it is that of the smallest snapshot id that is
+        //  larger than the requested one.
+        //
+        // In summary, we need to find an element in an array, returning the index of the smallest value that is larger if
+        // it is not found, unless said value doesn't exist (e.g. when all values are smaller). Arrays.findUpperBound does
+        // exactly this.
+
         uint256 index = snapshots.ids.findUpperBound(snapshotId);
```

Comment:
I recall you mentioning that the semantics of `findUpperBound` were very weird and very specific to `ERC20Snapshot`. What do you think about removing the `Arrays` library (that only contains `findUpperBound`) and moving the function to a private one in this contract.

Reply:
Yes, this is a good point. I wouldn't mind adding more functions to `Arrays` though, but I cannot think of any useful ones at the moment.

Something that slightly worries me is testing: we'd have to drop our extensive test suite for `findUpperBound` if we were to do this change as proposed.

Reply:
Yeah that's an interesting point... It's not that I worry about it being the only function in `Array`. I think `Array` could be a useful library. The issue I'm pointing out is that the semantics of `findUpperBound` are maybe not generally useful. We can open an issue to discuss this and tackle it before the release.

Reply:
Agreed. I created https://github.com/OpenZeppelin/openzeppelin-contracts/issues/2131